_includes/header.html

```diff
@@ -60,32 +60,154 @@
 	{% endcomment %}
 </header>
 
-<script>
-	document.addEventListener('click', function(event) {
-		const languageSelector = document.querySelector('.language-selector');
-		if (!languageSelector) return;
+{% if page.localize %}
+<script type="module">
+	let _languageSelector = null;
 
-		// Check if the click is outside the language selector
-		if (!languageSelector.contains(event.target)) {
-				languageSelector.classList.remove('open');
+	function isLanguageMatch(langA, langB, lax = false) {
+		const langALower = (langA ?? "").toLowerCase();
+		const langBLower = (langB ?? "").toLowerCase();
+		if (langALower === langBLower) {
+			return true;
+		}
+		if (!lax) {
+			return false;
+		}
+		return langALower.slice(0, 2) === langBLower.slice(0, 2);
+	}
+
+	function redirectTo(path) {
+		if (path == null) {
+			return;
 		}
-	});
-
-	function setLanguagePreference(event, element) {
-		event.preventDefault();
-		const path = element.getAttribute('data-lang-path');
-		const lang = element.getAttribute('data-lang');
-		
-		// Set cookie that expires in 365 days
-		const expirationDate = new Date();
-		expirationDate.setDate(expirationDate.getDate() + 365);
-		document.cookie = `preferred_language=${lang}; expires=${expirationDate.toUTCString()}; path=/; SameSite=Lax`;
-		
-		// Redirect to the language-specific path
 		window.location.href = path;
 	}
 
-	
+	function getLanguageSelector() {
+		if (_languageSelector != null) {
+			return _languageSelector;
+		}
+		_languageSelector = document.querySelector(".language-selector");
+		if (_languageSelector == null) {
+			throw new Error("Could not find `.language-selector`");
+		}
+		return _languageSelector;
+	}
+
+	function setPreferredLanguage(lang) {
+		window.localStorage.setItem("preferred_language", lang);
+	}
+
+	function getPreferredLanguage() {
+		return window.localStorage.getItem("preferred_language");
+	}
```

Comment:
Do you think we can use cookies here? We may want to redirect server side on initial page load, so it would be nice if we can access the preferred language there too.